inc/components/components/post-list/component.php

```diff
@@ -62,15 +63,23 @@
 			// Add the current $post_ids to the list of used ids.
 			post_list_get_and_add_used_post_ids( $post_ids );
 
-			// Ensure single items are wrapped in an array.
-			$item = ( isset( $templates['item'][0] ) ) ? $templates['item'] : [ $templates['item'] ];
+			// Track index.
+			$index = 0;
 
 			$children = array_map(
-				function ( $post_id ) use ( $item ) {
+				function ( $post_id ) use ( $templates, &$index ) {
+
+					// Decide which item to use.
+					$item = $templates['item_overrides'][ $index ] ?? $templates['item'];
+
+					// Ensure single items are wrapped in an array.
+					$item = ( isset( $item[0] ) ) ? $item : [ $item ];
+
 					return [
 						'name'     => 'irving/post-provider',
 						'config'   => [
 							'post_id' => $post_id,
+							'index'   => $index++,
```

Comment:
Is this going to make the index off by one compared to the position in the array? I would think that you would want to create the post-provider with the current index.

You could also do something like `array_map( callback_function( $item, $index ) {...}, $array, array_keys( $array ) )` to pass the index as the second param of the callback function. That said, a `foreach` would also be fine.

Reply:
Since it's a post increment operator, it returns $index, then adds one for the next loop around.

I'll look at the array keys option again though, I wasn't having luck earlier.